src/OrchardCore.Modules/OrchardCore.Security/Middlewares/PermissionsPolicyMiddleware.cs

```diff
@@ -0,0 +1,26 @@
+using System.Threading.Tasks;
+using Microsoft.AspNetCore.Http;
+using Microsoft.Extensions.Options;
+using OrchardCore.Security.Options;
+
+namespace OrchardCore.Security.Middlewares
+{
+    public class PermissionsPolicyMiddleware
+    {
+        private readonly PermissionsPolicyOptions _options;
+        private readonly RequestDelegate _next;
+
+        public PermissionsPolicyMiddleware(IOptions<PermissionsPolicyOptions> options, RequestDelegate next)
+        {
+            _options = options.Value;
+            _next = next;
+        }
+
+        public Task Invoke(HttpContext context)
+        {
+            context.Response.Headers[SecurityHeaderNames.PermissionsPolicy] = _options.ToString();
```

Comment:
Can't all these middleware be merged, and the code driven by the options?

Reply:
It could be merged as I suggested when I finished two or three headers, nothing but it could be useful from APIs perspective if someone want to use these outside OC

This module is using an extension method that calls others middlewares to inject their headers, it could be useful for readability and maintainability

Reply:
But it seems you still enlist multiples middlewares in the pipeline, just took a quick look so not sure but maybe only one middleware, maybe using some shared options with default values, and that would inject a kind of policy providers, whose registration may depend on enabled features, as you want.

Hmm, so maybe use the same kind of pattern used by the `RequestLocalizationMiddleware`.

Reply:
I merged them into a single one as I mentioned in my earlier commits. Thinking about prolicy providers idea